src/garage/torch/modules/multi_headed_mlp_module.py

```diff
@@ -0,0 +1,125 @@
+
+    Return:
+        The list of output torch. Tensor of the MLP
+    """
+    def __init__(self,
+                 n_heads,
+                 input_dim,
+                 output_dims,
+                 hidden_sizes,
+                 hidden_nonlinearity=nn.ReLU,
+                 hidden_w_init=nn.init.xavier_normal_,
+                 hidden_b_init=nn.init.zeros_,
+                 output_nonlinearities=None,
+                 output_w_inits=nn.init.xavier_normal_,
+                 output_b_inits=nn.init.zeros_,
+                 layer_normalization=False):
+        super().__init__()
+
+        self._layers = nn.ModuleList()
+
+        if n_heads < 1:
+            raise ValueError("n_head should be greater than or equal to 1 but {} provided.".format(n_heads))
+
+        if isinstance(output_dims, (list, tuple)):
+            if len(output_dims) != 1 and len(output_dims) != n_heads:
+                raise ValueError(self.ERROR_MESSAGE.format('output_dim', n_heads, output_dims))
+        else:
+            output_dims = [output_dims]
+
+        if isinstance(output_nonlinearities, (list, tuple)):
+            if len(output_nonlinearities) != 1 and len(output_nonlinearities) != n_heads:
+                raise ValueError(self.ERROR_MESSAGE.format('output_non_linearity', n_heads, output_nonlinearities))
+        else:
+            output_nonlinearities = [output_nonlinearities]
+
+        if isinstance(output_w_inits, (list, tuple)):
+            if len(output_w_inits) != 1 and len(output_w_inits) != n_heads:
+                raise ValueError(self.ERROR_MESSAGE.format('output_w_init', n_heads, output_w_inits))
+        else:
+            output_w_inits = [output_w_inits]
+
+        if isinstance(output_b_inits, (list, tuple)):
+            if len(output_b_inits) != 1 and len(output_b_inits) != n_heads:
+                raise ValueError(self.ERROR_MESSAGE.format('output_b_init', n_heads, output_b_inits))
+        else:
+            output_b_inits = [output_b_inits]
+
+        prev_size = input_dim
+        for size in hidden_sizes:
+            layer = nn.Sequential()
+            if layer_normalization:
+                layer.add_module("layer_normalization", nn.LayerNorm(prev_size))
+
+            linear_layer = nn.Linear(prev_size, size)
+            hidden_w_init(linear_layer.weight)
+            hidden_b_init(linear_layer.bias)
+            layer.add_module("linear", linear_layer)
+
+            if hidden_nonlinearity:
+                layer.add_module("non_linearity", hidden_nonlinearity)
+
+            self._layers.append(layer)
+            prev_size = size
+
+        self._output_layers = nn.ModuleList()
+        for i in range(n_heads):
+            layer = nn.Sequential()
+
+            output_dim = output_dims[0 if len(output_dims) == 1 else i]
+            linear_layer = nn.Linear(prev_size, output_dim)
+            output_w_inits[0 if len(output_w_inits) == 1 else i](linear_layer.weight)
```

Comment:
this is incredibly difficult to read. please see my suggestion above (just initialize it to be length n_heads)

Reply:
fixed